source/UF/SequentialColimits.lagda

```diff
@@ -0,0 +1,532 @@
+
+\begin{code}
+
+  gluing-from-sequential-cocone
+   : ((b , H) : sequential-cocone 𝓐 X)
+   → (c : Σ A + Σ A)
+   → b (index (f c)) (element-at (f c)) ＝ b (index (g c)) (element-at (g c))
+  gluing-from-sequential-cocone (b , H) (inl -) = refl
+  gluing-from-sequential-cocone (b , H) (inr (n , x)) = H n x
+
+  pushout-cocone-to-seq-cocone : cocone f g X → sequential-cocone 𝓐 X
+  pushout-cocone-to-seq-cocone (i , j , H) = (curry j , I)
+   where
+    I : (n : ℕ) → (curry j n) ∼ (λ - → j (succ n , a n -))
+    I n x = H (inl (n , x)) ⁻¹ ∙ H (inr (n , x))
+
+  seq-cocone-to-pushout-cocone : sequential-cocone 𝓐 X → cocone f g X
+  seq-cocone-to-pushout-cocone (b , H)
+   = (uncurry b , uncurry b , gluing-from-sequential-cocone (b , H))
+
+  pushout-cocone-to-seq-cocone-is-retraction
+   : pushout-cocone-to-seq-cocone ∘ seq-cocone-to-pushout-cocone ∼ id
+  pushout-cocone-to-seq-cocone-is-retraction (b , H)
+   = sequential-cocone-family-to-id 𝓐 X
+      (pushout-cocone-to-seq-cocone (seq-cocone-to-pushout-cocone (b , H)))
+      (b , H) ((λ n → λ x → refl) , (λ n → λ x → refl))
+
+  pushout-cocone-to-seq-cocone-is-section
+   : seq-cocone-to-pushout-cocone ∘ pushout-cocone-to-seq-cocone ∼ id
+  pushout-cocone-to-seq-cocone-is-section (i , j , H)
+   = inverse-cocone-map f g X
+      (seq-cocone-to-pushout-cocone (pushout-cocone-to-seq-cocone (i , j , H)))
+      (i , j , H) ((λ (n , x) → H (inl (n , x)) ⁻¹) , ∼-refl , I)
+   where
+    I : (z : Σ A + Σ A)
+      → H (inl (index (f z) , element-at (f z))) ⁻¹ ∙ H z
+      ＝ gluing-from-sequential-cocone
+         (curry j , λ n → λ x → H (inl (n , x)) ⁻¹ ∙ H (inr (n , x))) z
+    I (inl -) = left-inverse (H (inl -))
+    I (inr -) = refl
+
+  pushout-to-seq-cocone-is-equiv : is-equiv pushout-cocone-to-seq-cocone
+  pushout-to-seq-cocone-is-equiv
+   = qinvs-are-equivs pushout-cocone-to-seq-cocone
+      (seq-cocone-to-pushout-cocone ,
+       pushout-cocone-to-seq-cocone-is-section ,
+        pushout-cocone-to-seq-cocone-is-retraction)
+
+\end{code}
+
+Additionally, we show that the canonical map to sequential cocones factors through the
+canonical map to pushout cocones and the above map that translates between them.
+
+\begin{code}
+
+  canonical-maps-commute
+   : canonical-map-to-sequential-cocone 𝓐 sequential-colimit X
+      sequential-colimit-is-cocone
+   ∼ pushout-cocone-to-seq-cocone
+    ∘ canonical-map-to-cocone sequential-colimit f g pushout-cocone X
+  canonical-maps-commute u
+   = sequential-cocone-family-to-id 𝓐 X
+      (canonical-map-to-sequential-cocone 𝓐 sequential-colimit X
+       sequential-colimit-is-cocone u)
+      (pushout-cocone-to-seq-cocone (canonical-map-to-cocone sequential-colimit f g
+       pushout-cocone X u))
+      (I , II)
+    where
+     I : (n : ℕ) → u ∘ ι n ∼ curry (u ∘ inrr) n
+     I n x = refl
+     II : (n : ℕ) (x : A n)
+        → ap u (K n x)
+        ＝ refl ∙ (ap u (glue-inl (n , x)) ⁻¹ ∙ ap u (glue-inr (n , x)))
```

Comment:
You don't make use of `ap-glue-inl` here, so what did you intend to do?

Reply:
I think I just missed it